### PR TITLE
[new release] lacaml (11.0.10)

### PR DESCRIPTION
**Pull request by mmottl**
Lacaml - OCaml-bindings to BLAS and LAPACK

- Project page: <a href="https://mmottl.github.io/lacaml">https://mmottl.github.io/lacaml</a>
- Documentation: <a href="https://mmottl.github.io/lacaml/api">https://mmottl.github.io/lacaml/api</a>

##### CHANGES:

  * Fixed platform configuration bug

### CODE REVIEW

**Comment by mmottl** [comment]
@mseri It seems this latest patch still does not address the flag discovery problem on `ppc64-ocaml-5.0`.  Any idea on how this could be debugged more easily without access to the architecture?

**Comment by mseri** [comment]
In theory, if you have a linux machine with docker, this should use emulate ppc:
```
cd $(mktemp -d)
git clone --recursive "https://github.com/ocaml/opam-repository.git" && cd "opam-repository" && git fetch origin "refs/pull/23798/head" && git reset --hard 50eddc63
git fetch origin master
git merge --no-edit 97665ffaf0d03abf4361ba7ffb29ec6d7840b0e6
cat > ../Dockerfile <<'END-OF-DOCKERFILE'
FROM ocaml/opam:debian-11-ocaml-4.14@sha256:c120769c1842484248943ea52cb9e6764aa56bc35d4ca83b285d4c692d4f247d
USER 1000:1000
WORKDIR /home/opam
RUN sudo ln -f /usr/bin/opam-dev /usr/bin/opam
RUN opam init --reinit -ni
ENV OPAMDOWNLOADJOBS="1"
ENV OPAMERRLOGLEN="0"
ENV OPAMSOLVERTIMEOUT="500"
ENV OPAMPRECISETRACKING="1"
RUN rm -rf opam-repository/
COPY --chown=1000:1000 . opam-repository/
RUN opam repository set-url --strict default opam-repository/
RUN opam update --depexts || true
ENV OPAMCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMFIXUPCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMUPGRADECRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
RUN opam pin add -k version -yn lacaml.11.0.10 11.0.10
RUN opam reinstall lacaml.11.0.10; \
    res=$?; \
    test "$res" != 31 && exit "$res"; \
    export OPAMCLI=2.0; \
    build_dir=$(opam var prefix)/.opam-switch/build; \
    failed=$(ls "$build_dir"); \
    partial_fails=""; \
    for pkg in $failed; do \
    if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"debian-11\""; then \
    echo "A package failed and has been disabled for CI using the 'x-ci-accept-failures' field."; \
    fi; \
    test "$pkg" != 'lacaml.11.0.10' && partial_fails="$partial_fails $pkg"; \
    done; \
    test "${partial_fails}" != "" && echo "opam-repo-ci detected dependencies failing: ${partial_fails}"; \
    exit 1

END-OF-DOCKERFILE
docker build -f ../Dockerfile .
```

**Comment by mseri** [comment]
It should be ppc64 though: https://github.com/ocaml/opam/blob/f539e4c6fb00f3aabcb5ed10bbe1d1e49dd2abb7/src/state/opamSysPoll.ml#L30

**Comment by mseri** [comment]
In the logs there is a “ppc64el” appearing as the arch on debian’s packages… I wonder if we need to use that instead.

**Comment by avsm** [comment]
The ppc64 error on OCaml 5.0.0 is likely to be from a bytecode-only linking problem, since 5.x has not (yet) got ppc64 native code support yet.  If you try a bytecode only configuration of amd64 or arm64, it might trigger the same issue.

File: packages/lacaml/lacaml.11.0.10/opam [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/mmottl/lacaml.git"
```
Comment (mseri) [inline packages/lacaml/lacaml.11.0.10/opam:51]:
```suggestion
conflicts: [
  "ocaml-option-bytecode-only"
]
dev-repo: "git+https://github.com/mmottl/lacaml.git"
```

Comment (mseri) [reply packages/lacaml/lacaml.11.0.10/opam:51]: Thanks @avsm. Then this should fix the issue

**Comment by mseri** [comment]
ppc64 is still failing after declaring bytecode-only incompatibility :(

```

#=== ERROR while compiling lacaml.11.0.10 =====================================#
# context              2.2.0~alpha~dev | linux/ppc64 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/mmottl/lacaml/releases/download/11.0.10/lacaml-11.0.10.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/lacaml.11.0.10
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p lacaml -j 127 @install
# exit-code            1
# env-file             ~/.opam/log/lacaml-7-837c41.env
# output-file          ~/.opam/log/lacaml-7-837c41.out
### output ###
# File "src/config/dune", line 24, characters 11-34:
# 24 |     (names zdot_is_procedure_stubs zdot_is_function_stubs)
#                 ^^^^^^^^^^^^^^^^^^^^^^^
# (cd _build/default/src/config && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -DEXTERNAL_EXP10 -std=c99 -march=native -O3 -ffast-math -fPIC -DPIC -g -I /home/opam/.opam/4.14/lib/ocaml -o zdot_is_procedure_stubs.o -c zdot_is_procedure_stubs.c)
# gcc: error: unrecognized command-line option '-march=native'; did you mean '-mcpu=native'?
# File "src/config/dune", line 24, characters 35-57:
# 24 |     (names zdot_is_procedure_stubs zdot_is_function_stubs)
#                                         ^^^^^^^^^^^^^^^^^^^^^^
# (cd _build/default/src/config && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -DEXTERNAL_EXP10 -std=c99 -march=native -O3 -ffast-math -fPIC -DPIC -g -I /home/opam/.opam/4.14/lib/ocaml -o zdot_is_function_stubs.o -c zdot_is_function_stubs.c)
# gcc: error: unrecognized command-line option '-march=native'; did you mean '-mcpu=native'?
```

**Comment by mseri** [comment]
The libtriangle failure seems more an issue with ocaml 5, it can be fixed separately:
```
# libtriangle_stubs.c:64:35: error: invalid type argument of '->' (have 'int')
#    64 | #define VEC_OF_BA(ba) ((REAL *) ba->data)
#       |                                   ^~
```

**Comment by avsm** [comment]
This package is improving the OCaml 5 situation, so moving forward with merging it. Further portability fixes can be in a later release. Thanks @mmottl!

**Pull request merged**